### PR TITLE
Fixed Makie error in interactive_orbitdiagram

### PR DESCRIPTION
**Pull request by jonathanfischer97**
I fixed the `Makie.Layout` error I raised in #237 by removing that qualifier from the `deactivate_interaction!` call. 

### Other changes that can be ommitted upon review

- I removed some `display(figure)` calls because they seemed unnecessary and aren't shared by the other interactive plotting scripts. 
- I preallocated the `Vector{Point2f}` in the first `minimal_normalized_od` method because the length is known at compile time (the other method has conditional push so left it as is)
- Finally, I added a testset and a test for calling this plotting function. Not sure if you guys don't use a `Test.jl` based CI for a reason or was just never implemented. The test fails with an error with the prior code but passes now.

### CODE REVIEW

**Review by Datseris — COMMENTED** [review]
Thank you @jonathanfischer97 . The reason there are no tests here is because in the past DynamicalSystems.jl had **no source code**. It simple re-exported all other packages of the library. However, recently I decided to move here all GUI related code. However, that code never had tests because I didn't know exactly what or how to test plotting code.

Adding tests is of course an important and great improvement, so thank you for doing that. I left some comments to improve the tests. 

If by any chance you have some spare time, adding the same test `fig = gui_function(); @test fig isa Makie.figure()` to all GUI applications would be a great addition to the library.

Furthermore, can you please increase patch version number in Project.toml?

File: ext/src/orbitdiagram.jl [hunk 1]
```diff
@@ -4,14 +4,14 @@ function DynamicalSystems.interactive_orbitdiagram(ds, p_index, p_min, p_max, i0
...
-    display(figure)
+    figure = Figure(size = (1200, 600))
+    # display(figure)
     odax = figure[1,1] = Axis(figure; title)
     for z in (:xpanlock, :ypanlock, :xzoomlock, :yzoomlock)
         setproperty!(odax, z, true)
     end
     controllayout = figure[1, 2] = GridLayout()
-    display(figure)
+    # display(figure)
```
Comment (Datseris) [inline ext/src/orbitdiagram.jl:14]:
```suggestion
```

File: ext/src/orbitdiagram.jl [hunk 2]
```diff
@@ -4,14 +4,14 @@ function DynamicalSystems.interactive_orbitdiagram(ds, p_index, p_min, p_max, i0
         u0 = nothing, parname = "p", title = ""
     )
 
-    figure = Figure(resolution = (1200, 600))
-    display(figure)
+    figure = Figure(size = (1200, 600))
+    # display(figure)
```
Comment (Datseris) [inline ext/src/orbitdiagram.jl:8]:
```suggestion
```

File: ext/src/orbitdiagram.jl [hunk 3]
```diff
@@ -140,9 +140,11 @@ function minimal_normalized_od(ds, i, p_index, p₋, p₊,
 
     pvalues = range(p₋, stop = p₊, length = d)
     pdif = p₊ - p₋
-    od = Vector{Point2f}() # make this pre-allocated
+    # od = Vector{Point2f}() # make this pre-allocated
```
Comment (Datseris) [inline ext/src/orbitdiagram.jl:143]:
```suggestion
```

File: ext/src/orbitdiagram.jl [hunk 4]
```diff
@@ -151,13 +153,16 @@ function minimal_normalized_od(ds, i, p_index, p₋, p₊,
         for _ in 1:n
             step!(ds)
             x = current_state(ds)[i]
-            push!(od, Point2f(pp, x))
+            # push!(od, Point2f(pp, x))
```
Comment (Datseris) [inline ext/src/orbitdiagram.jl:156]:
```suggestion
```

**Review by Datseris — COMMENTED** [review]
no reason to leave commented code in

**Comment by jonathanfischer97** [comment]
No problem, happy to help! I'll add more tests as I explore more of the visualization methods!

**Pull request merged**